src/Pact/Compile.hs

```diff
@@ -70,7 +70,6 @@ type Compile a = ExpParse CompileState a
 initParseState :: Exp Info -> ParseState CompileState
```

Comment:
Oh man, I love `LambdaCase`. PRO arguments would be:

- It's a strict reduction of characters on the screen
- It avoids having to bind a name to something used only once
  - Corollary: the bound value can't accidentally be misused anywhere else, since it's gone

CON arguments would be:

- It's a bit funny to look at at first
- It's not strictly necessary (i.e. it's "sugar")
- It forces a pragma to be added at the top of the file (if pragma proliferation is a concern)

Reply:
To be clear, I have no objection to using `LambdaCase` in general, on a per-module basis. I do have an argument against it in that code maintenance can suffer when you find you do need to actually access the case value itself. But I think it is firmly module-governed; these modules don't use it, so don't change it.